### PR TITLE
Session.Send refactor

### PR DESCRIPTION
**Pull request by lucas-clemente**
To be merged after #694.

### CODE REVIEW

File: packet_packer.go [hunk 1]
```diff
@@ -185,9 +191,24 @@ func (p *packetPacker) composeNextPacket(
 	var payloadLength protocol.ByteCount
 	var payloadFrames []frames.Frame
 
+	// STOP_WAITING and ACK will always fit
```
Comment (marten-seemann) [inline packet_packer.go]: Until we implement timestamps in ACKs.

File: packet_packer.go [hunk 2]
```diff
@@ -185,9 +191,24 @@ func (p *packetPacker) composeNextPacket(
...
+		payloadFrames = append(payloadFrames, p.stopWaiting)
+		l, err := p.stopWaiting.MinLength(p.version)
+		if err != nil {
+			return nil, err
+		}
+		payloadLength += l
 	}
+	if p.ackFrame != nil {
+		payloadFrames = append(payloadFrames, p.ackFrame)
+		l, err := p.ackFrame.MinLength(p.version)
```
Comment (marten-seemann) [inline packet_packer.go]: Strictly speaking, this never errors for an ACK frame. I don't mind checking for the error here though for now, and removing the error from `MinLength` as soon as we finally drop the STOP_WAITING.

Comment (lucas-clemente) [inline packet_packer.go]: Yeah, I just don't like ignoring error values, as there's no compile-time check prohibiting you from e.g. changing AckFrame.MinLength to actually return an error.

File: packet_packer.go (hunk 1 above)
Comment (lucas-clemente) [inline packet_packer.go]: Good point, but then ACK frame writing will become much harder anyway (probably with the need to pass in a max len to Write()).

**Pull request merged**